cuqi/geometry/_geometry.py

```diff
@@ -732,7 +733,28 @@ def par2fun(self, p):
 
     def fun2par(self, funvals):
         """The function to parameter map used to map function values back to parameters, if available."""
-        raise NotImplementedError("fun2par not implemented. ")
+        # Check that the input is of the correct shape
+        if len(funvals) != self.fun_dim:
+            raise ValueError(
+                "Input array funvals must have length {}".format(self.fun_dim))
+        
+        warnings.warn(
+            f"fun2par for {self.__class__} is a projection on "
+            + "the KL expansion coefficients space where only "
+            + f"the first self.num_modes={self.num_modes} "
+            + "coefficients are returned. "
+            + _disable_warning_msg("cuqi.geometry")
+            + "\n"
+        )
+
+
+        # Note, the scaling by 2*self.fun_dim is not needed if scipy.
```

Comment:
It's good to mention the points of this comment. In general before giving technicalities, first explain what the purpose of the following code is and how the code achieves that purpose. Here specifically, we want to explain that we compute a projection onto the coefficient space etc. and why the dst and the rest of the commands produce that (maybe give a reference to where this expression comes from).

Reply:
Makes sense, I added more explanation to clarify this point.